### PR TITLE
clean transitions between struct-nested fields

### PR DESCRIPTION
**Pull request by bmschmidt**
There was a bug caused by a failure to properly sort values in the list of buffers. When delving into struct array fields, an additional sort operation is necessary.

This fixes that, and also includes a number of type improvements that I made while hunting down that bug. 

----

> [!IMPORTANT]
> Fixes sorting issue in `SwitchPositions.svelte` for struct fields and improves type safety with `TupleMap` across the codebase.
> 
>   - **Behavior**:
>     - Fixes sorting issue in `SwitchPositions.svelte` by ensuring nested struct fields are handled correctly.
>     - Updates `click()` function to handle struct fields with dynamic subfields.
>   - **Type Improvements**:
>     - Replaces `Record<string, number>` with `TupleMap<string, number>` in `Deeptable.ts`, `AestheticSet.ts`, and `regl_rendering.ts` for better type safety.
>     - Updates `TupleMap` and `TupleSet` usage in `utilityFunctions.ts`.
>   - **Misc**:
>     - Removes unused code and comments in `SwitchPositions.svelte` and `selection.ts`.
>     - Minor logging and code cleanup in `regl_rendering.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=nomic-ai%2Fdeepscatter&utm_source=github&utm_medium=referral)<sup> for 5e2f32a0eb04cd14336f8580cc5c11e3403d10b7. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by bmschmidt** [comment]
* **#160** <a href="https://app.graphite.dev/github/pr/nomic-ai/deepscatter/160?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bmschmidt and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: dev/svelte/SwitchPositions.svelte [hunk 1]
```diff
@@ -13,75 +13,61 @@
...
-      scatterplot.deeptable.transformations['struct' + i] = async function (
-        tile,
-      ) {
-        // Create a nested struct with a change.
-        const x = (await tile.get_column('x')).toArray();
-        const y = (await tile.get_column('y')).toArray();
+    if (scatterplot.deeptable.transformations['struct'] === undefined) {
+      scatterplot.deeptable.transformations['struct'] = async function (tile) {
+        const fields = [];
+        const vectors = [];
```
Comment (bmschmidt) [inline dev/svelte/SwitchPositions.svelte:19]: This test is changed to use a nested struct.

File: src/regl_rendering.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export class ReglRenderer extends Renderer {
   public reglframe?: REGL.Cancellable;
   public bufferManager: BufferManager;
 
-  private aes_to_buffer_num?: Record<string, number>;
+  private aes_to_buffer_num: TupleMap<string, number> = new TupleMap();
```
Comment (bmschmidt) [inline src/regl_rendering.ts:70]: Type of this changed from Record to TupleMap

File: src/regl_rendering.ts [hunk 3]
```diff
@@ -946,27 +945,37 @@ export class ReglRenderer extends Renderer {
...
+        aes_to_buffer_num.set(k, variable_to_buffer_num.get(field));
         continue;
       }
       if (num++ < 16) {
-        aes_to_buffer_num[k] = num;
+        aes_to_buffer_num.set(k, num);
         variable_to_buffer_num.set(field, num);
         continue;
       } else {
+        console.log('Overflow');
```
Comment (RLesser) [inline src/regl_rendering.ts:959]: remove?

File: src/regl_rendering.ts [hunk 4]
```diff
@@ -946,27 +945,37 @@ export class ReglRenderer extends Renderer {
...
         // Don't use the last value, use the current value.
         // Loses animation but otherwise plots nicely.
-        // Strategy will break if more than 15 base channels are defined,
+        // Strategy will break if more than 14 base channels are defined,
```
Comment (RLesser) [inline src/regl_rendering.ts:962]: interesting

File: src/regl_rendering.ts [hunk 5]
```diff
@@ -1055,6 +1064,7 @@ export class BufferManager {
             if (tile.readyToUse) {
               // tile.get_column(keyset[0]);
             } else {
+              // pass
```
Comment (RLesser) [inline src/regl_rendering.ts:1067]: fine that nothing happens in either case?

**Review by RLesser — APPROVED** [review]
LGTM

**Comment by bmschmidt** [comment]
### Merge activity

* **Oct 16, 1:45 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/nomic-ai/deepscatter/160).

**Pull request merged**